### PR TITLE
build: fix 'make lint' when GOPATH isn't explicitly set

### PR DESCRIPTION
**Pull request by dcbw**
@squeed @girishmg

### CODE REVIEW

File: go-controller/hack/lint.sh [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #!/usr/bin/env bash
+set -x
```
Comment (girishmg) [inline go-controller/hack/lint.sh]: @dcbw do we need this `set -x`? the build is becoming noisy? thanks.

File: go-controller/hack/lint.sh [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
 
-GO111MODULE=on golangci-lint run \
+GOPATH=${GOPATH:-$(go env GOPATH)}
+curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b ${GOPATH}/bin
```
Comment (squeed) [inline go-controller/hack/lint.sh]: I'd really rather not execute this every time I lint...

File: go-controller/hack/lint.sh (hunk 1 above)
Comment (dcbw) [reply go-controller/hack/lint.sh]: removed

File: go-controller/hack/lint.sh (hunk 2 above)
Comment (dcbw) [reply go-controller/hack/lint.sh]: now checks for that file before downloading it again. But didn't your original PR do the same thing too, or is Make smart enough to not re-curl the file somehow?

**Comment by dcbw** [comment]
@girishmg @squeed PTAL thanks

**Comment by girishmg** [comment]
@dcbw /lgtm

File: go-controller/hack/lint.sh [hunk 3]
```diff
@@ -1,6 +1,10 @@
 #!/usr/bin/env bash
 
-GO111MODULE=on golangci-lint run \
+GOPATH=${GOPATH:-$(go env GOPATH)}
+if [[ ! -x "${GOPATH}/bin/golangci-lint" ]]; then
+  curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b ${GOPATH}/bin
```
Comment (girishmg) [inline go-controller/hack/lint.sh]: @dcbw how do we now go to the latest release of golanci-lint? don't we be stuck with the current version of v1.20.1?

Comment (dcbw) [reply go-controller/hack/lint.sh]: @girishmg good question; but I'm not sure how the original Makefile one got the latest either? Either that or it always downloaded it, which is what Casey said he didnt want.

So I really don't know what I'm supposed to do here. Either we always download the linter, or we only download it once.

(If Casey is talking about the 'go env GOPATH' well I disagree that running it every time is a problem, and he can just set GOPATH then)

File: go-controller/hack/lint.sh (hunk 2 above)
Comment (dcbw) [reply go-controller/hack/lint.sh]: @squeed do you mean the curl or the `go env GOPATH`?

**Comment by dcbw** [comment]
OK, reverted to mostly what Casey has, but we now fix GOPATH up in the Makefile itself. PTAL thanks.

**Comment by dcbw** [comment]
@squeed @girishmg PTAL thanks!

**Comment by girishmg** [comment]
@dcbw LGTM

**Pull request merged**